meta-xt-control-domain/recipes-guest/doma/files/doma-create-ExecStopPost.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Note!
+#
+# Below we are writing xen-storage status BEFORE executing destruction
+# of the DomA Xen domain. That is done by intention in order to allow
+# QEMU to be destroyed ASAP.
+#
+# At current moment, we are accessing '/dev/mmcblk0p3' device using
+# both, 'PV block' and 'virtio' virtual devices. That is done,
+# because AOSP uses MMC as a 'virtio' device shared by QEMU, while
+# during the U-Boot phase a 'PV block' device shared by Xen is being
+# used.
+#
+# Writing "dead" status to "drivers/dom0-qemu-command-monitor/status"
+# of DomD will cause qemu-command-handler service in DomD to be
+# stopped. That will unblock shutdown of the DomA guest domain.
+
+DOMD_ID=$(xl list | awk '{ if ($1 == "DomD") print $2 }') && \
+/usr/bin/xenstore-write /local/domain/${DOMD_ID}/drivers/dom0-qemu-command-monitor/status dead && \
+echo "Destroying DomA" && ( /usr/sbin/xl -v destroy DomA || true ) && echo "Destroyed DomA"; \
+
+# Let's wait for hanging domains to shutdown
+HANGING_DOMAIN_ID_MEMORIZED="";
+
+while true; do
+    HANGING_DOMAIN_ID=$(xl list | awk '{ if ($1 == "(null)") print $2 }');
```

Comment:
Hi @lorc, We have DOMD_ID here. But we are destroying DomA.

After DomA is destroyed, it often happens that it is still stuck in "xl list" with its name equal to "(null)". Probably domain is releasing its resources. I don't know for sure. It can happen for dozens of seconds.

Here I'm waiting for the domain to totally disappear from the "xl list" because otherwise, it might happen that (null) domain still consumes memory, and the creation of a new instance of DomA will fail in such case. It actually has happened.

Also, it could happen, that when I reach this script the domain was already destroyed ( e.g. internal reboot of the domain ). So getting its ID here was not an option.

That's why it is so tricky.